### PR TITLE
Add script to make testing on forks easier

### PR DESCRIPTION
**Pull request by natalieparellano**
Tested on [fork](https://github.com/natalieparellano/lifecycle/actions/runs/1140708853), ha! 

Signed-off-by: Natalie Arellano <narellano@vmware.com>

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -27,6 +27,43 @@
...
+echo "Use public key from fork (assumes private key and passphrase have been added to GitHub secrets)"
+cp ~/workspace/fork.pub cosign.pub
+
+echo "Use own Docker Hub account (assumes docker password has been added to GitHub secrets)"
+sed -i '' "s/buildpacksio\/lifecycle/$1\/lifecycle/g" ~/workspace/lifecycle/.github/workflows/build.yml
+sed -i '' "s/buildpacksio\/lifecycle/$1\/lifecycle/g" ~/workspace/lifecycle/.github/workflows/post-release.yml
+
+echo "Skip tests to make things faster"
+sed -i '' "s/make test/echo "test"/g" ~/workspace/lifecycle/.github/workflows/build.yml
+sed -i '' "s/make acceptance/echo "acceptance"/g" ~/workspace/lifecycle/.github/workflows/build.yml
```
Comment (natalieparellano) [inline DEVELOPMENT.md:65]:
```suggestion
sed -i '' "s/make test/echo test/g" ~/workspace/lifecycle/.github/workflows/build.yml
sed -i '' "s/make acceptance/echo acceptance/g" ~/workspace/lifecycle/.github/workflows/build.yml
```

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -27,6 +27,43 @@
...
+echo "Use public key from fork (assumes private key and passphrase have been added to GitHub secrets)"
+cp ~/workspace/fork.pub cosign.pub
+
+echo "Use own Docker Hub account (assumes docker password has been added to GitHub secrets)"
+sed -i '' "s/buildpacksio\/lifecycle/$1\/lifecycle/g" ~/workspace/lifecycle/.github/workflows/build.yml
+sed -i '' "s/buildpacksio\/lifecycle/$1\/lifecycle/g" ~/workspace/lifecycle/.github/workflows/post-release.yml
+
+echo "Skip tests to make things faster"
+sed -i '' "s/make test/echo test/g" ~/workspace/lifecycle/.github/workflows/build.yml
+sed -i '' "s/make acceptance/echo acceptance/g" ~/workspace/lifecycle/.github/workflows/build.yml
```
Comment (yaelharel) [inline DEVELOPMENT.md:65]: Can we just remove the sections of `pack-acceptance` so we won't even set anything before not running the acceptance tests?

**Review by yaelharel — COMMENTED** [review]
This script is great! It'll make testing on fork so much more convenient!
Thanks for creating it!!!

I added a small suggestion.

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -27,6 +27,43 @@
...
+
+The following script can be used to update the source code to reflect the state of the fork. Note that it assumes the lifecycle is cloned inside `~/workspace/` and that there is a cosign public key saved at `~/workspace/fork.pub`.
+It can be invoked like so: `<path to script> <Docker Hub username> <GitHub username>`
+
+```bash
+# $1 - Docker Hub username
+# $2 - GitHub username
+
+echo "Use fork"
+sed -i '' "s/\"buildpacks\"/\"$2\"/g" ~/workspace/lifecycle/.github/workflows/draft-release.yml
```
Comment (yaelharel) [inline DEVELOPMENT.md:54]: Probably this is redundant following https://github.com/buildpacks/lifecycle/pull/695.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -27,6 +27,43 @@
...
+### Testing GitHub actions on forks
+
+The lifecycle release process involves chaining a series of GitHub actions together such that:
+* The "build" workflow creates the artifacts
+  * .tgz files containing the lifecycle binaries, shasums for the .tgz files, a cosign public key, an SBOM, etc.
+  * OCI images containing the lifecycle binaries, tagged with their commit sha (for more information, see RELEASE.md)
+* The "draft-release" workflow finds the artifacts and downloads them, creating the draft release
+* The "post-release" workflow re-tags the OCI images that were created during the "build" workflow with the release version
+
+It can be rather cumbersome to test changes to these workflows, as they are heavily intertwined. Thus we recommend forking the buildpacks/lifecycle repository in GitHub and running through the entire release process end-to-end.
```
Comment (imjasonh) [inline DEVELOPMENT.md:39]: I had good luck just updating image refs to points to ghcr.io/$gh-username/lifecycle, and changing the docker-login steps to login using `${{ secrets.GITHUB_TOKEN }}`. This meant I didn't have to set the `DOCKER_*` secrets, which was kinda nice.

**Review by imjasonh — COMMENTED** [review]
Thanks for documenting and improving this workflow!

File: DEVELOPMENT.md (hunk 4 above)
Comment (natalieparellano) [reply DEVELOPMENT.md:39]: Thanks for the suggestion! I updated the script to accommodate this use case.

File: DEVELOPMENT.md (hunk 2 above)
Comment (natalieparellano) [reply DEVELOPMENT.md]: I thought about this, but then if we decided to add steps that depend on pack acceptance passing, we'd have to remember to update the script... IDK though, maybe the time savings now would be worth it.

Comment (natalieparellano) [reply DEVELOPMENT.md]: Decided to remove pack-acceptance-* for now (it saves about a minute, so the total execution time is around 5 minutes). Thanks for the suggestion!

**Pull request merged**